cicecore/cicedyn/dynamics/ice_transport_remap.F90

```diff
@@ -1939,13 +1965,34 @@ subroutine locate_triangles (nx_block,     ny_block,   &
          ishift_bc =  0
          jshift_bc =  0
 
+         ! index shifts for neighbor north edges
+
+         ise_tl =  1
+         jse_tl =  0
+         ise_bl =  0
+         jse_bl =  0
+         ise_tr =  1
+         jse_tr = -1
+         ise_br =  0
+         jse_br = -1
+
          ! area scale factors
 
          do j = jb, je
          do i = ib, ie
-            areafac_l(i,j) = dxu(i,j  )*dyu(i,j  )
-            areafac_r(i,j) = dxu(i,j-1)*dyu(i,j-1)
-            areafac_c(i,j) = p5 * (areafac_l(i,j) + areafac_r(i,j))
+            areafac_l = dxu(i,j  )*dyu(i,j  )
+            areafac_r = dxu(i,j-1)*dyu(i,j-1)
+            areafac_c(i,j) = p5*( areafac_l + areafac_r )
+         enddo
+         enddo
+
+         ! area scale factor for other edge (north)
+
+         do j = jb-1, je
+         do i = ib, ie+1
+            areafac_l = dxu(i-1,j)*dyu(i-1,j)
+            areafac_r = dxu(i  ,j)*dyu(i  ,j)
+            areafac_ce(i,j) = p5*( areafac_l + areafac_r )
```

Comment:
What's happening here.  Why do you have a 2nd loop that overwrites most of the first loop.  Also, @TillRasmussen is correct, these indices are problematic.  Is there confusion about ib=ilo?  I'd love to see ib and ie removed and use ilo and ihi with explicit offsets instead to make things clearer.

Reply:
It does not overwrite it. The first array is areafac_c the other one is areafac_ce.

Reply:
Good point, areafac_l and areafac_r are recomputed though.